### PR TITLE
Resolves cast error parsing 7-zip entries

### PR DESCRIPTION
**Pull request by idlira**
PropID.LAST_MODIFICATION_TIME returns a Date (or null) which can't be parsed as Long.

Therefore we use getTime or assume the Instant.now in the unlikely event where the modification date isn't present in the archive

Fixes: SIRI-373

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/util/SevenZipAdapter.java [hunk 1]
```diff
@@ -82,13 +83,16 @@ public ISequentialOutStream getStream(int index, ExtractAskMode extractAskMode)
...
-        currentLastModified = Instant.ofEpochSecond((Long) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME));
 
         // Enforce the filename filter...
         if (!filter.test(currentFilePath)) {
             return null;
         }
 
+        Date lastModificationTime = (Date) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME);
+        currentLastModified =
+                lastModificationTime == null ? Instant.now() : Instant.ofEpochSecond(lastModificationTime.getTime());
```
Comment (meg1502) [inline src/main/java/sirius/biz/util/SevenZipAdapter.java:94]: `Optional.ofNullable(lastModificationTime).map(date -> Instant.ofEpochSecond(date.getTime())).orElse(Instant.now());`

typing it out i realize your version is probably cleaner..

**Review by meg1502 — APPROVED** [review]

File: src/main/java/sirius/biz/util/SevenZipAdapter.java (hunk 1 above)
Comment (fhaScireum) [reply src/main/java/sirius/biz/util/SevenZipAdapter.java:94]: I think its cleaner with optional chain as then we don't have two similiar named date variables in scope.

**Review by fhaScireum — APPROVED** [review]

File: src/main/java/sirius/biz/util/SevenZipAdapter.java (hunk 1 above)
Comment (sabieber) [reply src/main/java/sirius/biz/util/SevenZipAdapter.java]: Wollts auch fast vorschlagen :D

File: src/main/java/sirius/biz/util/SevenZipAdapter.java [hunk 2]
```diff
@@ -82,13 +84,16 @@ public ISequentialOutStream getStream(int index, ExtractAskMode extractAskMode)
...
-        currentLastModified = Instant.ofEpochSecond((Long) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME));
 
         // Enforce the filename filter...
         if (!filter.test(currentFilePath)) {
             return null;
         }
 
+        currentLastModified = Optional.ofNullable((Date) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME))
+                                      .map(lastModificationTime -> Instant.ofEpochMilli(lastModificationTime.getTime()))
+                                      .orElse(Instant.now());
```
Comment (sabieber) [inline src/main/java/sirius/biz/util/SevenZipAdapter.java:95]:
```suggestion
        currentLastModified = Optional.ofNullable((Date) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME))
                                      .map(Date::getTime)
                                      .map(Instant::ofEpochMilli)
                                      .orElse(Instant.now());
```

**Review by ymo-sci — APPROVED** [review]

**Review by meg1502 — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**